File: client/Attension/AttentionItem.swift
//
//  AttentionItem.swift
//  Attension
//
//  Created by Hiraku Ohno on 2016/04/13.
//  Copyright © 2016年 Hiraku Ohno. All rights reserved.
//

import UIKit
import RealmSwift

class AttentionItem: Object {
    dynamic var identifier: String = ""
    dynamic var latitude: CLLocationDegrees = 0.0
    dynamic var longtitude: CLLocationDegrees = 0.0
    dynamic var attentionBody: String = ""
    dynamic var placeName: String = ""
    dynamic var shared: Bool = false
    
    override class func primaryKey() -> String? {
        return "identifier"
    }
}

extension AttentionItem {
    /* onServer should be in response only. */
    func toDictionary(includeIdentifier: Bool = true) -> [String : AnyObject] {
        var ret: [String : AnyObject] = ["latitude"       : latitude,
                                         "longitude"      : longtitude,
                                         "attention_body" : attentionBody,
                                         "place_name"     : placeName]

        if includeIdentifier {
            ret["identifier"] = identifier
        }
        return ret
    }
}

func ==(left: AttentionItem, right: AttentionItem) -> Bool {
    return left.identifier == right.identifier
}

func !=(left: AttentionItem, right: AttentionItem) -> Bool {
    return !(right == left)
}

File: server/attention_item.py
__author__ = 'hira'

class AttentionItem(object) :

    def __init__(self):
        self._identifier = ''
        self._place_name = ''
        self._attention_body = ''
        self._latitude = 0
        self._longitude = 0

    @property
    def identifier(self):
        return self._identifier

    @identifier.setter
    def identifier(self, identifier):
        self._identifier = identifier

    @property
    def place_name(self):
        return self._place_name

    @place_name.setter
    def place_name(self, place_name):
        self._place_name = place_name

    @property
    def attention_body(self):
        return self._attention_body

    @attention_body.setter
    def attention_body(self, attention_body):
        self._attention_body = attention_body

    @property
    def latitude(self):
        return self._latitude

    @latitude.setter
    def latitude(self, latitude):
        self._latitude = latitude

    @property
    def longitude(self):
        return self._longitude

    @longitude.setter
    def longitude(self, longitude):
        self._longitude = longitude

